aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -352,23 +330,20 @@ void cpu_flash_attention(
           tmp_max = qk_max_data[row] > tmp_max ? qk_max_data[row] : tmp_max;
           // qk <- exp(qk - max) and sum per row
           tmp_sum = tmp_max;
-          _exp_reduce_sum_fusion_kernel(qk_data + row * kvBlockSize, kvBlockSize, qk_data + row * kvBlockSize, tmp_sum);
+          _exp_reduce_sum_fusion_kernel(
+              qk_data + row * kvBlockSize, kvBlockSize,
+              conditional_data_ptr(qk_data, qk_reduced_data) + row * kvBlockSize,
```

Comment:
One noob question about this conditional_data_ptr: TORCH_INTERNAL_ASSERT is only enabled for debug builds, which are not shipped via pip or conda. I am not even sure if they are run in CI/CD. Should this be a runtime check with `TORCH_CHECK`?

Reply:
Yes, `TORCH_CHECK` should be better. @Valentine233

Reply:
Thanks! Would fix this in future PRs.

Reply:
Thanks! Have fixed this.